include/seqan3/argument_parser/validators.hpp

```diff
@@ -485,6 +485,98 @@ struct default_validator
     }
 };
 
+//!\brief A helper struct to chain validators recursively via operator+()
+//!\ingroup validators
+template <validator_concept validator1_type, validator_concept validator2_type>
+    requires std::Same<typename validator1_type::value_type, typename validator2_type::value_type>
+class validator_chain_adaptor
+{
+    //!\brief The underlying type in both validators.
+    using value_type = typename validator1_type::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \brief The (default) constructors.
+     * \{
+     */
+    //!\brief The default constructor is explicitly deleted.
+    validator_chain_adaptor() = delete;
+    validator_chain_adaptor(validator_chain_adaptor const & pf) = default;
+    validator_chain_adaptor & operator=(validator_chain_adaptor const & pf) = default;
+    validator_chain_adaptor(validator_chain_adaptor &&) = default;
+    validator_chain_adaptor & operator=(validator_chain_adaptor &&) = default;
+
+    /*!\brief Constructing from two validators.
+     * \param[in] vali1_ Some validator to be chained to vali2_.
+     * \param[in] vali2_ Another validator to be chained to vali1_.
+     */
+    validator_chain_adaptor(validator1_type const & vali1_, validator2_type const & vali2_) :
```

Comment:
Are validator movable? If so take them by copy and move them into the object parameter